src/main.ts

```diff
@@ -789,6 +796,16 @@ function generateEncode(ctx: Context, fullName: string, messageDesc: DescriptorP
             ${writeSnippet('v!')};
           }
         `);
+      } else if (isEnum(field) && options.stringEnums) {
+        const tag = ((field.number << 3) | basicWireType(field.type)) >>> 0;
```

Comment:
@kanziw I'm pretty sure that `| basicWriteType(...)` is wrong here, because we're encoding the repeated field as a packed repeated field, so that tag should just be `| 2`:

https://developers.google.com/protocol-buffers/docs/encoding

I've made this change in my #284 version of this fix, but let me know if I'm missing something.